src/test/java/org/apache/commons/codec/net/URLCodecTest.java

```diff
@@ -129,14 +130,19 @@ public void testEncodeDecodeNull() throws Exception {
 
 
     @Test
-    public void testDecodeInvalid() throws Exception {
+    public void testDecodeInvalid() {
         final URLCodec urlCodec = new URLCodec();
-        assertThrows(DecoderException.class, () -> urlCodec.decode("%"));
-        assertThrows(DecoderException.class, () -> urlCodec.decode("%A"));
-        // Bad 1st char after %
-        assertThrows(DecoderException.class, () -> urlCodec.decode("%A"));
-        // Bad 2nd char after %
-        assertThrows(DecoderException.class, () -> urlCodec.decode("%0W"));
+
+        assertAll(
+                () -> assertThrows(DecoderException.class, () -> urlCodec.decode("%")),
+
+                // Bad 1st char after %
+                () -> assertThrows(DecoderException.class, () -> urlCodec.decode("%A")),
```

Comment:
Hm, isn't it missing one call from the existing code above? Using `%A`?

Reply:
So old line 135 and 137 appear to be the exact same... so that got merged into new line 140.

Reply:
I thought that was intentional to verify some behavior with a duplicate input, but didn't look into the code.